docs/architecture/adr-023-protobuf-naming.md

```diff
@@ -0,0 +1,238 @@
+the best choices for the future.
+
+### Versioning
+
+#### Don't Allow Breaking Changes in Stable Packages
+
+Always use a breaking change detector such as [Buf](https://buf.build) to prevent
+breaking changes in stable (non-alpha or beta) packages. Breaking changes can
+break smart contracts/persistent scripts and generally provide a bad UX for
+clients. With protobuf, there should usually be ways to extend existing
+functionality instead of just breaking it.
+
+#### Use Simple Version Suffixes
+
+Instead of using [Buf's recommended version suffix](https://buf.build/docs/lint-checkers#package_version_suffix),
+we can more concisely indicate version 2, 3, etc. of a package with a simple
+package suffix rather than a sub-package, ex. `cosmos.bank2.Send` or `cosmos.bank.Send2`.
+Version 1 should obviously be the non-suffixed version, ex. `cosmos.bank.Send`.
+
+#### Use `alpha` or `beta` to Denote Non-stable Packages
+
+[Buf's recommended version suffix](https://buf.build/docs/lint-checkers#package_version_suffix)
+(ex. `v1alpha1`) _should_ be used for non-stable packages. These packages should
+likely be excluded from breaking change detection and _should_ generally 
+be blacklisted from usage by smart contracts/persistent scripts to prevent them
+from breaking. The SDK _should_ mark any packages as alpha or beta where the
+API is likely to change significantly in the near future.
+
+### Package Naming
+
+#### Adopt a short, unique top-level package name
+
+Top-level packages should adopt a short name that is known to not collide with
+other names in common usage within the Cosmos ecosystem. In the near future, a
+registry should be created to reserve and index top-level package names used
+within the Cosmos ecosystem. Because the Cosmos SDK is intended to provide
+the top-level types for the Cosmos project, the top-level package name `cosmos`
+is recommended for usage within the Cosmos SDK instead of the longer `cosmos_sdk`.
+[ICS](https://github.com/cosmos/ics) specifications could consider a
+short top-level package like `ics23` based upon the standard number.
+
+#### Limit sub-package depth
+
+Sub-package depth should be increased with caution. Generally a single
+sub-package is needed for a module or a library. Even though `x` or `modules`
+is used in source code to denote modules, this is often unnecessary for .proto
+files as modules are the primary thing sub-packages are used for. Only items which
+are known to be used infrequently should have deep sub-package depths.
+
+For the Cosmos SDK, it is recommended that that we simply write `cosmos.bank`,
+`cosmos.gov`, etc. rather than `cosmos.x.bank`. In practice, most non-module
+types can go straight in the `cosmos` package or we can introduce a
+`cosmos.base` package if needed. Note that this naming _will not_ change
+go package names, i.e. the `cosmos.bank` protobuf package will still live in
+`x/bank`.
+
+### Message Naming
+
+#### Use simple verbs for transaction messages
```

Comment:
I have to disagree with this section. I'd rather opt to include the `Msg` prefix as I think it adds just enough context at very little cost.